src/integrationtests/java/com/aws/greengrass/integrationtests/KeystoreTest.java

```diff
@@ -37,21 +44,52 @@
 import static com.aws.greengrass.lifecyclemanager.GreengrassService.SERVICES_NAMESPACE_TOPIC;
 import static com.aws.greengrass.testcommons.testutilities.ExceptionLogProtector.ignoreExceptionOfType;
 import static com.aws.greengrass.testcommons.testutilities.TestUtils.asyncAssertOnConsumer;
+import static com.github.grantwest.eventually.EventuallyLambdaMatcher.eventuallyEval;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @BridgeIntegrationTest
 public class KeystoreTest {
     private static final long AWAIT_TIMEOUT_SECONDS = 30L;
 
+    ExecutorService executor = Executors.newCachedThreadPool();
     BridgeIntegrationTestContext testContext;
 
+    @AfterEach
+    void tearDown() {
+        executor.shutdownNow();
+    }
+
+    @TestWithMqtt5Broker
+    @WithKernel("mqtt5_config_ssl.yaml")
+    void GIVEN_mqtt_bridge_WHEN_client_cert_changes_THEN_local_client_restarts(Broker broker) throws Exception {
```

Comment:
Does this test prove the client does not work as it is today (without the client cert listener)?

Reply:
yep, test fails if `reset` is removed from the client